libraries/AP_Math/AP_Math.h

```diff
@@ -25,273 +17,115 @@
-	return ((amt)<(low)?(low):((amt)>(high)?(high):(amt)));
-}
-
-//matrix algebra
-bool inverse(float x[], float y[], uint16_t dim);
-
-// degrees -> radians
-static inline float radians(float deg) {
-	return deg * DEG_TO_RAD;
-}
-
-// radians -> degrees
-static inline float degrees(float rad) {
-	return rad * RAD_TO_DEG;
+template <class T>
+auto wrap_180_cd(const T &angle) -> decltype(wrap_180(angle, 100.f)) {
+    return wrap_180(angle, 100.f);
 }
 
-// square
-static inline float sq(float v) {
-	return v*v;
-}
-
-// 2D vector length
-static inline float pythagorous2(float a, float b) {
-	return sqrtf(sq(a)+sq(b));
-}
-
-// 3D vector length
-static inline float pythagorous3(float a, float b, float c) {
-	return sqrtf(sq(a)+sq(b)+sq(c));
-}
-
-#ifdef radians
-#error "Build is including Arduino base headers"
-#endif
-
+/* 
+ * @brief: Gets two values and returns the smaller one.
+ */
 template<typename A, typename B>
-static inline auto MIN(const A &one, const B &two) -> decltype(one < two ? one : two) {
+auto MIN(const A &one, const B &two) -> decltype(one < two ? one : two) { 
     return one < two ? one : two;
 }
 
+/* 
+ * @brief: Gets two values and returns the bigger one.
+ */
 template<typename A, typename B>
-static inline auto MAX(const A &one, const B &two) -> decltype(one > two ? one : two) {
+auto MAX(const A &one, const B &two) -> decltype(one > two ? one : two) {  
     return one > two ? one : two;
 }
 
-#define NSEC_PER_SEC 1000000000ULL
-#define NSEC_PER_USEC 1000ULL
-#define USEC_PER_SEC 1000000ULL
-
-inline uint32_t hz_to_nsec(uint32_t freq)
-{
-    return NSEC_PER_SEC / freq;
+/*
+ * Converter functions
+ *  - Avoid zero divisions
+ *  - Inheritss a float cast (because of PX4)
+ */
+template<class T>
+T hz_to_nsec(const T &freq) {
+    T val = !is_zero(static_cast<float>(freq)) ? NSEC_PER_SEC / freq : 0;
+    return val;
 }
 
-inline uint32_t nsec_to_hz(uint32_t nsec)
-{
-    return NSEC_PER_SEC / nsec;
+template<class T>
+T nsec_to_hz(const T &nsec) {
+    T val = !is_zero(static_cast<float>(nsec)) ? NSEC_PER_SEC / nsec : 0;
+    return val;
 }
 
-inline uint32_t usec_to_nsec(uint32_t usec)
-{
-    return usec * NSEC_PER_USEC;
+template<class T>
+T usec_to_nsec(const T &usec) {
+    T val = usec * NSEC_PER_USEC;
+    return val;
 }
 
-inline uint32_t nsec_to_usec(uint32_t nsec)
-{
-    return nsec / NSEC_PER_USEC;
+template<class T>
+T nsec_to_usec(const T &nsec) {
+    T val = !is_zero(static_cast<float>(nsec)) ? nsec / NSEC_PER_USEC : 0;
+    return val;
 }
 
-inline uint32_t hz_to_usec(uint32_t freq)
-{
-    return USEC_PER_SEC / freq;
+template<class T>
+T hz_to_usec(const T &freq) {
+    T val = !is_zero(static_cast<float>(freq)) ? USEC_PER_SEC / freq : 0;
+    return val;
 }
 
-inline uint32_t usec_to_hz(uint32_t usec)
-{
-    return USEC_PER_SEC / usec;
+template<class T>
+T usec_to_hz(const T &usec) {
+    T val = !is_zero(static_cast<float>(usec)) ? USEC_PER_SEC / usec : 0;
+    return val;
 }
```

Comment:
same comment for all these functions... they are not supposed to use floating point math.